src/lib/Form/Data/Search/TrashSearchData.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\Data\Search;
+
+use eZ\Publish\API\Repository\Values\Content\Section;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\API\Repository\Values\User\User;
+use Symfony\Component\Validator\Constraints as Assert;
+
+class TrashSearchData
+{
+    /**
+     * @var int|null
+     *
+     * @Assert\Range(
+     *     max = 500
+     * )
+     */
+    private $limit;
+
+    /** @var int|null */
+    private $page;
+
+    /** @var \eZ\Publish\API\Repository\Values\Content\Section|null */
+    private $section;
+
+    /** @var \eZ\Publish\API\Repository\Values\ContentType\ContentType|null */
+    private $contentType;
+
+    /** @var array|null */
+    private $trashedInterval;
+
+    /** @var string|null */
+    private $trashed;
+
+    /** @var \eZ\Publish\API\Repository\Values\User\User|null */
+    private $creator;
+
+    /** @var array|null */
+    private $sort;
+
+    /**
+     * @param int|null $limit
+     * @param int|null $page
+     * @param \eZ\Publish\API\Repository\Values\Content\Section|null $section
+     * @param \eZ\Publish\API\Repository\Values\ContentType\ContentType|null $contentType
+     * @param string|null $trashed
+     * @param array|null $trashedInterval
+     * @param \eZ\Publish\API\Repository\Values\User\User|null $creator
+     */
```

Comment:
all those phpdocs are not needed but those with array typehint but those needs to be specified